web/pgadmin/tools/psql/__init__.py

```diff
@@ -531,50 +561,22 @@ def get_connection_status(conn):
     return False
 
 
-def _get_database(sid, did):
+def _get_database_role(sid, did):
     """
     This method is used to get database based on sid, did.
     """
     try:
         from pgadmin.utils.driver import get_driver
         manager = get_driver(PG_DEFAULT_DRIVER).connection_manager(int(sid))
-        conn = manager.connection()
-        db_name = 'postgres'
+        conn = manager.connection(did=int(did))
 
         is_connected = get_connection_status(conn)
 
-        if is_connected:
-
-            if conn.manager and conn.manager.db_info \
-                    and conn.manager.db_info[int(did)] is not None:
-
-                db_name = conn.manager.db_info[int(did)]['datname']
-                return db_name
-            elif sid:
```

Comment:
Any specific reason to remove the else?

Reply:
I discussed this with Khusboo and she suggested else part would never be executed so removed.

Reply:
The code written was not necessary. We can get the database name from the connection itself; there is no need for a server manager.